lib/routes/account.js

```diff
@@ -457,47 +466,22 @@ module.exports = function (
             .then(
               function (result) {
                 emailRecord = result
-
-                allowSigninUnblock = features.isSigninUnblockEnabledForUser(emailRecord.uid, email, request)
-                if (allowSigninUnblock && unblockCode) {
-                  return db.consumeUnblockCode(emailRecord.uid, unblockCode)
-                    .then(
-                      (code) => {
-                        if (Date.now() - code.createdAt > unblockCodeLifetime) {
-                          log.info({
-                            op: 'Account.login.unblockCode.expired',
-                            uid: emailRecord.uid.toString('hex')
-                          })
-                          throw error.invalidUnblockCode()
-                        }
-                        didSigninUnblock = true
-                        return request.emitMetricsEvent('account.login.confirmedUnblockCode')
-                      }
-                    )
-                    .catch(
-                      (err) => {
-                        if (err.errno === error.ERRNO.INVALID_UNBLOCK_CODE) {
-                          request.emitMetricsEvent('account.login.invalidUnblockCode')
-                          customs.flag(request.app.clientAddress, {
-                            email: email,
-                            errno: err.errno
-                          })
-                        }
-                        throw err
-                      }
-                    )
-                }
-                if (!didSigninUnblock && customsErr) {
-                  throw customsErr
-                }
```

Comment:
I pulled all of this out into a separate function so it's easier to find and follow.